process/block/shardblock.go

```diff
@@ -681,9 +684,13 @@ func (sp *shardProcessor) CommitBlock(
 		log.Debug(errNotCritical.Error())
 	}
 
+	log.Info(fmt.Sprintf("shardBlock with nonce %d is the highest block notarized by metachain for shard %d\n",
+		sp.forkDetector.GetHighestFinalBlockNonce(),
+		sp.shardCoordinator.SelfId()))
+
 	sp.appStatusHandler.SetStringValue(core.MetricCurrentBlockHash, core.ToB64(headerHash))
 
-	hdrsToAttestFinality := uint32(header.Nonce - finalHeaders[0].GetNonce())
+	hdrsToAttestFinality := uint32(header.Nonce - sp.forkDetector.GetHighestFinalBlockNonce() + 1)
 	sp.removeNotarizedHdrsBehindFinal(hdrsToAttestFinality)
```

Comment:
rename as suggested in case of metablock

Reply:
done